src/Suite.ts

```diff
@@ -1,52 +1,99 @@
 /* eslint-env node, mocha */
-import assert from 'assert';
-import each from 'lodash/each';
+import defer from 'lodash/defer';
 import groupBy from 'lodash/groupBy';
-import options from './Options';
+import { createEmitter, Emitter } from './Emitter';
+import { Options } from './Options';
 import * as ProcessedStory from './ProcessedStory';
 import * as Result from './Result';
 import * as TestBrowser from './TestBrowser';
 
 /**
- * Find Storybook stories and generate a test for each one.
+ * Mapping of event names to handlers for the test suite.
  */
-async function writeTests() {
-  const testBrowser = await TestBrowser.create(options);
-  const page = await TestBrowser.createPage(testBrowser, options);
-  const stories = await TestBrowser.getStories(page);
-  const storiesByComponent = groupBy(stories, 'componentTitle');
-
-  describe(`[${options.browser}] accessibility`, function () {
-    after(async function () {
+export type SuiteEvents = {
+  suiteStart: (browser: string) => void;
+  componentStart: (componentName: string) => void;
+  componentSkip: (componentName: string) => void;
+  storyStart: (storyName: string) => void;
+  storyPass: (storyName: string, result: Result.Result, elapsedTime: number) => void;
+  storyFail: (storyName: string, result: Result.Result, elapsedTime: number) => void;
+  storySkip: (storyName: string) => void;
+  suiteFinish: (numPass: number, numFail: number, numSkip: number, elapsedTime: number) => void;
+}
+
+/**
+ * Event emitter representing a run of the test suite.
+ */
+export type SuiteEmitter = Emitter<SuiteEvents>;
+
+/**
+ * Find Storybook stories and run Axe on each one. Returns an event emitter that emits events when
+ * components and stories are processed.
+ */
+export function run(options: Options): SuiteEmitter {
```

Comment:
This file is now undoubtedly uglier 😑 .

That's because Mocha was doing the heavy lifting and formatting for us. So we've removed a heavy dependency on Mocha, and traded it for more complexity here (and in the formatting code). Let me know if you think it's worth it or not.

We could try to implement our own `describe`, `it`, etc, to clean this up a bit.

Reply:
I think the tradeoff is ok 👍  Are we still enforcing a timeout / should we do it here?

Reply:
We're not enforcing a timeout. Good idea to do that. Will have to think about how to accomplish that 🤔